// @radixdlt/babylon-gateway-api-sdk/dist/babylon-gateway-api-sdk.mjs
var ke = Object.defineProperty;
var Me = (e, t, r) => t in e ? ke(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r;
var S = (e, t, r) => (Me(e, typeof t != "symbol" ? t + "" : t, r), r);
const Le = "https://rcnet.radixdlt.com".replace(/\/+$/, "");
class Ke {
  constructor(t = {}) {
    this.configuration = t;
  }
  set config(t) {
    this.configuration = t;
  }
  get basePath() {
    return this.configuration.basePath != null ? this.configuration.basePath : Le;
  }
  get fetchApi() {
    return this.configuration.fetchApi;
  }
  get middleware() {
    return this.configuration.middleware || [];
  }
  get queryParamsStringify() {
    return this.configuration.queryParamsStringify || z;
  }
  get username() {
    return this.configuration.username;
  }
  get password() {
    return this.configuration.password;
  }
  get apiKey() {
    const t = this.configuration.apiKey;
    if (t)
      return typeof t == "function" ? t : () => t;
  }
  get accessToken() {
    const t = this.configuration.accessToken;
    if (t)
      return typeof t == "function" ? t : async () => t;
  }
  get headers() {
    return this.configuration.headers;
  }
  get credentials() {
    return this.configuration.credentials;
  }
}
const Ge = new Ke();
class b {
  constructor(t = Ge) {
    S(this, "middleware");
    S(this, "fetchApi", async (t, r) => {
      let i = { url: t, init: r };
      for (const o of this.middleware)
        o.pre && (i = await o.pre({
          fetch: this.fetchApi,
          ...i
        }) || i);
      let s;
      try {
        s = await (this.configuration.fetchApi || fetch)(i.url, i.init);
      } catch (o) {
        for (const a of this.middleware)
          a.onError && (s = await a.onError({
            fetch: this.fetchApi,
            url: i.url,
            init: i.init,
            error: o,
            response: s ? s.clone() : void 0
          }) || s);
        if (s === void 0)
          throw o instanceof Error ? new ze(o, "The request failed and the interceptors did not return an alternative response") : o;
      }
      for (const o of this.middleware)
        o.post && (s = await o.post({
          fetch: this.fetchApi,
          url: i.url,
          init: i.init,
          response: s.clone()
        }) || s);
      return s;
    });
    this.configuration = t, this.middleware = t.middleware;
  }
  withMiddleware(...t) {
    const r = this.clone();
    return r.middleware = r.middleware.concat(...t), r;
  }
  withPreMiddleware(...t) {
    const r = t.map((i) => ({ pre: i }));
    return this.withMiddleware(...r);
  }
  withPostMiddleware(...t) {
    const r = t.map((i) => ({ post: i }));
    return this.withMiddleware(...r);
  }
  async request(t, r) {
    const { url: i, init: s } = await this.createFetchParams(t, r), o = await this.fetchApi(i, s);
    if (o.status >= 200 && o.status < 300)
      return o;
    throw new Be(o, "Response returned an error code");
  }
  async createFetchParams(t, r) {
    let i = this.configuration.basePath + t.path;
    t.query !== void 0 && Object.keys(t.query).length !== 0 && (i += "?" + this.configuration.queryParamsStringify(t.query));
    const s = Object.assign({}, this.configuration.headers, t.headers);
    Object.keys(s).forEach((B) => s[B] === void 0 ? delete s[B] : {});
    const o = typeof r == "function" ? r : async () => r, a = {
      method: t.method,
      headers: s,
      body: t.body,
      credentials: this.configuration.credentials
    }, y = {
      ...a,
      ...await o({
        init: a,
        context: t
      })
    }, Ve = {
      ...y,
      body: $e(y.body) || y.body instanceof URLSearchParams || Ue(y.body) ? y.body : JSON.stringify(y.body)
    };
    return { url: i, init: Ve };
  }
  clone() {
    const t = this.constructor, r = new t(this.configuration);
    return r.middleware = this.middleware.slice(), r;
  }
}
function Ue(e) {
  return typeof Blob < "u" && e instanceof Blob;
}
function $e(e) {
  return typeof FormData < "u" && e instanceof FormData;
}
class Be extends Error {
  constructor(r, i) {
    super(i);
    S(this, "name", "ResponseError");
    this.response = r;
  }
}
class ze extends Error {
  constructor(r, i) {
    super(i);
    S(this, "name", "FetchError");
    this.cause = r;
  }
}
class d extends Error {
  constructor(r, i) {
    super(i);
    S(this, "name", "RequiredError");
    this.field = r;
  }
}
const ti = {
  csv: ",",
  ssv: " ",
  tsv: "	",
  pipes: "|"
};
function n(e, t) {
  const r = e[t];
  return r != null;
}
function z(e, t = "") {
  return Object.keys(e).map((r) => W(r, e[r], t)).filter((r) => r.length > 0).join("&");
}
function W(e, t, r = "") {
  const i = r + (r.length ? `[${e}]` : e);
  if (t instanceof Array) {
    const s = t.map((o) => encodeURIComponent(String(o))).join(`&${encodeURIComponent(i)}=`);
    return `${encodeURIComponent(i)}=${s}`;
  }
  if (t instanceof Set) {
    const s = Array.from(t);
    return W(e, s, r);
  }
  return t instanceof Date ? `${encodeURIComponent(i)}=${encodeURIComponent(t.toISOString())}` : t instanceof Object ? z(t, i) : `${encodeURIComponent(i)}=${encodeURIComponent(String(t))}`;
}
function ni(e, t) {
  return Object.keys(e).reduce(
    (r, i) => ({ ...r, [i]: t(e[i]) }),
    {}
  );
}
function ri(e) {
  for (const t of e)
    if (t.contentType === "multipart/form-data")
      return !0;
  return !1;
}
class u {
  constructor(t, r = (i) => i) {
    this.raw = t, this.transformer = r;
  }
  async value() {
    return this.transformer(await this.raw.json());
  }
}
class ii {
  constructor(t) {
    this.raw = t;
  }
  async value() {
  }
}
class si {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.blob();
  }
}
class oi {
  constructor(t) {
    this.raw = t;
  }
  async value() {
    return await this.raw.text();
  }
}
function ai(e) {
  return !0;
}
function _(e) {
  return We(e);
}
function We(e, t) {
  return e == null ? e : {
    state_version: n(e, "state_version") ? e.state_version : void 0,
    timestamp: n(e, "timestamp") ? e.timestamp === null ? null : new Date(e.timestamp) : void 0,
    epoch: n(e, "epoch") ? e.epoch : void 0,
    round: n(e, "round") ? e.round : void 0
  };
}
function f(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      timestamp: e.timestamp === void 0 ? void 0 : e.timestamp === null ? null : e.timestamp.toISOString(),
      epoch: e.epoch,
      round: e.round
    };
}
function ui(e) {
  return !0;
}
function li(e) {
  return He(e);
}
function He(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0
  };
}
function ci(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state)
    };
}
function di(e) {
  let t = !0;
  return t = t && "value" in e, t;
}
function Qe(e) {
  return Xe(e);
}
function Xe(e, t) {
  return e == null ? e : {
    value: e.value,
    address: n(e, "address") ? e.address : void 0
  };
}
function Ye(e) {
  if (e !== void 0)
    return e === null ? null : {
      value: e.value,
      address: e.address
    };
}
const _i = {
  Unknown: "Unknown",
  CommittedSuccess: "CommittedSuccess",
  CommittedFailure: "CommittedFailure",
  Pending: "Pending",
  Rejected: "Rejected"
};
function T(e) {
  return Ze(e);
}
function Ze(e, t) {
  return e;
}
function fi(e) {
  return e;
}
function gi(e) {
  let t = !0;
  return t = t && "state_version" in e, t = t && "transaction_status" in e, t;
}
function E(e) {
  return je(e);
}
function je(e, t) {
  return e == null ? e : {
    state_version: e.state_version,
    transaction_status: T(e.transaction_status),
    payload_hash_hex: n(e, "payload_hash_hex") ? e.payload_hash_hex : void 0,
    intent_hash_hex: n(e, "intent_hash_hex") ? e.intent_hash_hex : void 0,
    fee_paid: n(e, "fee_paid") ? Qe(e.fee_paid) : void 0,
    confirmed_at: n(e, "confirmed_at") ? e.confirmed_at === null ? null : new Date(e.confirmed_at) : void 0,
    error_message: n(e, "error_message") ? e.error_message : void 0
  };
}
function D(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version,
      transaction_status: e.transaction_status,
      payload_hash_hex: e.payload_hash_hex,
      intent_hash_hex: e.intent_hash_hex,
      fee_paid: Ye(e.fee_paid),
      confirmed_at: e.confirmed_at === void 0 ? void 0 : e.confirmed_at === null ? null : e.confirmed_at.toISOString(),
      error_message: e.error_message
    };
}
function mi(e) {
  return !0;
}
function pi(e) {
  return et(e);
}
function et(e, t) {
  return e == null ? e : {
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0
  };
}
function yi(e) {
  if (e !== void 0)
    return e === null ? null : {
      cursor: e.cursor,
      limit_per_page: e.limit_per_page
    };
}
function Si(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "raw_json" in e, t;
}
function tt(e) {
  return nt(e);
}
function nt(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    raw_json: e.raw_json,
    as_string: n(e, "as_string") ? e.as_string : void 0,
    as_string_collection: n(e, "as_string_collection") ? e.as_string_collection : void 0
  };
}
function rt(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      raw_json: e.raw_json,
      as_string: e.as_string,
      as_string_collection: e.as_string_collection
    };
}
function Oi(e) {
  let t = !0;
  return t = t && "key" in e, t = t && "value" in e, t = t && "last_updated_at_state_version" in e, t;
}
function A(e) {
  return it(e);
}
function it(e, t) {
  return e == null ? e : {
    key: e.key,
    value: tt(e.value),
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function q(e) {
  if (e !== void 0)
    return e === null ? null : {
      key: e.key,
      value: rt(e.value),
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Ni(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function H(e) {
  return st(e);
}
function st(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(A)
  };
}
function Q(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(q)
    };
}
function Ri(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Fi(e) {
  return ot(e);
}
function ot(e, t) {
  return e == null ? e : {
    items: e.items.map(A)
  };
}
function hi(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(q)
    };
}
function bi(e) {
  return !0;
}
function Ti(e) {
  return at(e);
}
function at(e, t) {
  return e == null ? e : {
    as_string: n(e, "as_string") ? e.as_string : void 0,
    as_string_collection: n(e, "as_string_collection") ? e.as_string_collection : void 0
  };
}
function Ji(e) {
  if (e !== void 0)
    return e === null ? null : {
      as_string: e.as_string,
      as_string_collection: e.as_string_collection
    };
}
function Ii(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function ut(e) {
  return g(e, !1);
}
function g(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.type === "EntityNotFoundError")
      return X(e, !0);
    if (e.type === "InternalServerError")
      return re(e, !0);
    if (e.type === "InvalidEntityError")
      return ie(e, !0);
    if (e.type === "InvalidRequestError")
      return ae(e, !0);
    if (e.type === "InvalidTransactionError")
      return ue(e);
    if (e.type === "NotSyncedUpError")
      return me(e, !0);
    if (e.type === "TransactionNotFoundError")
      return Pe(e, !0);
  }
  return {
    type: e.type
  };
}
function m(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
function wi(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function vi(e) {
  return X(e, !1);
}
function X(e, t) {
  return e == null ? e : {
    ...g(e, t),
    address: e.address
  };
}
function Ei(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      address: e.address
    };
}
function Di(e) {
  let t = !0;
  return t = t && "message" in e, t;
}
function Ai(e) {
  return lt(e);
}
function lt(e, t) {
  return e == null ? e : {
    message: e.message,
    code: n(e, "code") ? e.code : void 0,
    details: n(e, "details") ? ut(e.details) : void 0,
    trace_id: n(e, "trace_id") ? e.trace_id : void 0
  };
}
function qi(e) {
  if (e !== void 0)
    return e === null ? null : {
      message: e.message,
      code: e.code,
      details: m(e.details),
      trace_id: e.trace_id
    };
}
const Pi = {
  Global: "Global",
  Vault: "Vault"
};
function p(e) {
  return ct(e);
}
function ct(e, t) {
  return e;
}
function xi(e) {
  return e;
}
function Ci(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t;
}
function P(e) {
  return x(e, !1);
}
function x(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.aggregation_level === "Global")
      return Y(e, !0);
    if (e.aggregation_level === "Vault")
      return ee(e, !0);
  }
  return {
    aggregation_level: p(e.aggregation_level),
    resource_address: e.resource_address
  };
}
function N(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address
    };
}
function Vi(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function dt(e) {
  return _t(e);
}
function _t(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(P)
  };
}
function ft(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(N)
    };
}
function ki(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Mi(e) {
  return gt(e);
}
function gt(e, t) {
  return e == null ? e : {
    items: e.items.map(P)
  };
}
function Li(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(N)
    };
}
function Ki(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function J(e) {
  return Y(e, !1);
}
function Y(e, t) {
  return e == null ? e : {
    ...x(e, t),
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function I(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...N(e),
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Gi(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Ui(e) {
  return mt(e);
}
function mt(e, t) {
  return e == null ? e : {
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function $i(e) {
  if (e !== void 0)
    return e === null ? null : {
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Bi(e) {
  let t = !0;
  return t = t && "vault_address" in e, t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function C(e) {
  return pt(e);
}
function pt(e, t) {
  return e == null ? e : {
    vault_address: e.vault_address,
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function V(e) {
  if (e !== void 0)
    return e === null ? null : {
      vault_address: e.vault_address,
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function zi(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Z(e) {
  return yt(e);
}
function yt(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(C)
  };
}
function j(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(V)
    };
}
function Wi(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function Hi(e) {
  return ee(e, !1);
}
function ee(e, t) {
  return e == null ? e : {
    ...x(e, t),
    vaults: Z(e.vaults)
  };
}
function Qi(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...N(e),
      vaults: j(e.vaults)
    };
}
function Xi(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function Yi(e) {
  return St(e);
}
function St(e, t) {
  return e == null ? e : {
    vaults: Z(e.vaults)
  };
}
function Zi(e) {
  if (e !== void 0)
    return e === null ? null : {
      vaults: j(e.vaults)
    };
}
function ji(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function es(e) {
  return Ot(e);
}
function Ot(e, t) {
  return e == null ? e : {
    items: e.items.map(C)
  };
}
function ts(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(V)
    };
}
function ns(e) {
  let t = !0;
  return t = t && "state_version" in e, t;
}
function rs(e) {
  return Nt(e);
}
function Nt(e, t) {
  return e == null ? e : {
    state_version: e.state_version
  };
}
function is(e) {
  if (e !== void 0)
    return e === null ? null : {
      state_version: e.state_version
    };
}
function ss(e) {
  let t = !0;
  return t = t && "release_version" in e, t = t && "open_api_schema_version" in e, t;
}
function te(e) {
  return Rt(e);
}
function Rt(e, t) {
  return e == null ? e : {
    release_version: e.release_version,
    open_api_schema_version: e.open_api_schema_version
  };
}
function ne(e) {
  if (e !== void 0)
    return e === null ? null : {
      release_version: e.release_version,
      open_api_schema_version: e.open_api_schema_version
    };
}
function os(e) {
  let t = !0;
  return t = t && "network" in e, t = t && "state_version" in e, t = t && "proposer_round_timestamp" in e, t = t && "epoch" in e, t = t && "round" in e, t;
}
function l(e) {
  return Ft(e);
}
function Ft(e, t) {
  return e == null ? e : {
    network: e.network,
    state_version: e.state_version,
    proposer_round_timestamp: e.proposer_round_timestamp,
    epoch: e.epoch,
    round: e.round
  };
}
function c(e) {
  if (e !== void 0)
    return e === null ? null : {
      network: e.network,
      state_version: e.state_version,
      proposer_round_timestamp: e.proposer_round_timestamp,
      epoch: e.epoch,
      round: e.round
    };
}
function as(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "release_info" in e, t;
}
function ht(e) {
  return bt(e);
}
function bt(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    release_info: te(e.release_info)
  };
}
function us(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      release_info: ne(e.release_info)
    };
}
function ls(e) {
  let t = !0;
  return t = t && "release_info" in e, t;
}
function cs(e) {
  return Tt(e);
}
function Tt(e, t) {
  return e == null ? e : {
    release_info: te(e.release_info)
  };
}
function ds(e) {
  if (e !== void 0)
    return e === null ? null : {
      release_info: ne(e.release_info)
    };
}
function _s(e) {
  let t = !0;
  return t = t && "exception" in e, t = t && "cause" in e, t;
}
function fs(e) {
  return re(e, !1);
}
function re(e, t) {
  return e == null ? e : {
    ...g(e, t),
    exception: e.exception,
    cause: e.cause
  };
}
function gs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      exception: e.exception,
      cause: e.cause
    };
}
function ms(e) {
  let t = !0;
  return t = t && "exception" in e, t = t && "cause" in e, t;
}
function ps(e) {
  return Jt(e);
}
function Jt(e, t) {
  return e == null ? e : {
    exception: e.exception,
    cause: e.cause
  };
}
function ys(e) {
  if (e !== void 0)
    return e === null ? null : {
      exception: e.exception,
      cause: e.cause
    };
}
function Ss(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function Os(e) {
  return ie(e, !1);
}
function ie(e, t) {
  return e == null ? e : {
    ...g(e, t),
    address: e.address
  };
}
function Ns(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      address: e.address
    };
}
function Rs(e) {
  let t = !0;
  return t = t && "path" in e, t = t && "errors" in e, t;
}
function se(e) {
  return It(e);
}
function It(e, t) {
  return e == null ? e : {
    path: e.path,
    errors: e.errors
  };
}
function oe(e) {
  if (e !== void 0)
    return e === null ? null : {
      path: e.path,
      errors: e.errors
    };
}
function Fs(e) {
  let t = !0;
  return t = t && "validation_errors" in e, t;
}
function hs(e) {
  return ae(e, !1);
}
function ae(e, t) {
  return e == null ? e : {
    ...g(e, t),
    validation_errors: e.validation_errors.map(se)
  };
}
function bs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      validation_errors: e.validation_errors.map(oe)
    };
}
function Ts(e) {
  let t = !0;
  return t = t && "validation_errors" in e, t;
}
function Js(e) {
  return wt(e);
}
function wt(e, t) {
  return e == null ? e : {
    validation_errors: e.validation_errors.map(se)
  };
}
function Is(e) {
  if (e !== void 0)
    return e === null ? null : {
      validation_errors: e.validation_errors.map(oe)
    };
}
function ws(e) {
  return !0;
}
function vs(e) {
  return ue(e);
}
function ue(e, t) {
  return e;
}
function Es(e) {
  return e;
}
function Ds(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function As(e) {
  return vt(e);
}
function vt(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state)
  };
}
function qs(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
function Ps(e) {
  let t = !0;
  return t = t && "faucet" in e, t = t && "epoch_manager" in e, t = t && "clock" in e, t = t && "ecdsa_secp256k1" in e, t = t && "eddsa_ed25519" in e, t = t && "xrd" in e, t;
}
function Et(e) {
  return Dt(e);
}
function Dt(e, t) {
  return e == null ? e : {
    faucet: e.faucet,
    epoch_manager: e.epoch_manager,
    clock: e.clock,
    ecdsa_secp256k1: e.ecdsa_secp256k1,
    eddsa_ed25519: e.eddsa_ed25519,
    xrd: e.xrd
  };
}
function At(e) {
  if (e !== void 0)
    return e === null ? null : {
      faucet: e.faucet,
      epoch_manager: e.epoch_manager,
      clock: e.clock,
      ecdsa_secp256k1: e.ecdsa_secp256k1,
      eddsa_ed25519: e.eddsa_ed25519,
      xrd: e.xrd
    };
}
function xs(e) {
  let t = !0;
  return t = t && "network_id" in e, t = t && "network_name" in e, t = t && "well_known_addresses" in e, t;
}
function qt(e) {
  return Pt(e);
}
function Pt(e, t) {
  return e == null ? e : {
    network_id: e.network_id,
    network_name: e.network_name,
    well_known_addresses: Et(e.well_known_addresses)
  };
}
function Cs(e) {
  if (e !== void 0)
    return e === null ? null : {
      network_id: e.network_id,
      network_name: e.network_name,
      well_known_addresses: At(e.well_known_addresses)
    };
}
const Vs = {
  String: "String",
  Integer: "Integer",
  Bytes: "Bytes",
  Uuid: "Uuid"
};
function w(e) {
  return xt(e);
}
function xt(e, t) {
  return e;
}
function ks(e) {
  return e;
}
function Ms(e) {
  let t = !0;
  return t = t && "non_fungible_id" in e, t;
}
function k(e) {
  return Ct(e);
}
function Ct(e, t) {
  return e == null ? e : {
    non_fungible_id: e.non_fungible_id
  };
}
function M(e) {
  if (e !== void 0)
    return e === null ? null : {
      non_fungible_id: e.non_fungible_id
    };
}
function Ls(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function le(e) {
  return Vt(e);
}
function Vt(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(k)
  };
}
function ce(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(M)
    };
}
function Ks(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Gs(e) {
  return kt(e);
}
function kt(e, t) {
  return e == null ? e : {
    items: e.items.map(k)
  };
}
function Us(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(M)
    };
}
function $s(e) {
  let t = !0;
  return t = t && "aggregation_level" in e, t = t && "resource_address" in e, t;
}
function L(e) {
  return K(e, !1);
}
function K(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.aggregation_level === "Global")
      return de(e, !0);
    if (e.aggregation_level === "Vault")
      return ge(e, !0);
  }
  return {
    aggregation_level: p(e.aggregation_level),
    resource_address: e.resource_address
  };
}
function R(e) {
  if (e !== void 0)
    return e === null ? null : {
      aggregation_level: e.aggregation_level,
      resource_address: e.resource_address
    };
}
function Bs(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Mt(e) {
  return Lt(e);
}
function Lt(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(L)
  };
}
function Kt(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(R)
    };
}
function zs(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Ws(e) {
  return Gt(e);
}
function Gt(e, t) {
  return e == null ? e : {
    items: e.items.map(L)
  };
}
function Hs(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(R)
    };
}
function Qs(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Xs(e) {
  return de(e, !1);
}
function de(e, t) {
  return e == null ? e : {
    ...K(e, t),
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function Ys(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...R(e),
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function Zs(e) {
  let t = !0;
  return t = t && "amount" in e, t = t && "last_updated_at_state_version" in e, t;
}
function js(e) {
  return Ut(e);
}
function Ut(e, t) {
  return e == null ? e : {
    amount: e.amount,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function eo(e) {
  if (e !== void 0)
    return e === null ? null : {
      amount: e.amount,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function to(e) {
  let t = !0;
  return t = t && "vault_address" in e, t = t && "total_count" in e, t = t && "last_updated_at_state_version" in e, t;
}
function G(e) {
  return $t(e);
}
function $t(e, t) {
  return e == null ? e : {
    vault_address: e.vault_address,
    total_count: e.total_count,
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function U(e) {
  if (e !== void 0)
    return e === null ? null : {
      vault_address: e.vault_address,
      total_count: e.total_count,
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function no(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function _e(e) {
  return Bt(e);
}
function Bt(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(G)
  };
}
function fe(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(U)
    };
}
function ro(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function io(e) {
  return ge(e, !1);
}
function ge(e, t) {
  return e == null ? e : {
    ...K(e, t),
    vaults: _e(e.vaults)
  };
}
function so(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...R(e),
      vaults: fe(e.vaults)
    };
}
function oo(e) {
  let t = !0;
  return t = t && "vaults" in e, t;
}
function ao(e) {
  return zt(e);
}
function zt(e, t) {
  return e == null ? e : {
    vaults: _e(e.vaults)
  };
}
function uo(e) {
  if (e !== void 0)
    return e === null ? null : {
      vaults: fe(e.vaults)
    };
}
function lo(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function co(e) {
  return Wt(e);
}
function Wt(e, t) {
  return e == null ? e : {
    items: e.items.map(G)
  };
}
function _o(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(U)
    };
}
function fo(e) {
  let t = !0;
  return t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function go(e) {
  return me(e, !1);
}
function me(e, t) {
  return e == null ? e : {
    ...g(e, t),
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
  };
}
function mo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
    };
}
function po(e) {
  let t = !0;
  return t = t && "request_type" in e, t = t && "current_sync_delay_seconds" in e, t = t && "max_allowed_sync_delay_seconds" in e, t;
}
function yo(e) {
  return Ht(e);
}
function Ht(e, t) {
  return e == null ? e : {
    request_type: e.request_type,
    current_sync_delay_seconds: e.current_sync_delay_seconds,
    max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
  };
}
function So(e) {
  if (e !== void 0)
    return e === null ? null : {
      request_type: e.request_type,
      current_sync_delay_seconds: e.current_sync_delay_seconds,
      max_allowed_sync_delay_seconds: e.max_allowed_sync_delay_seconds
    };
}
const Oo = {
  EcdsaSecp256k1: "EcdsaSecp256k1",
  EddsaEd25519: "EddsaEd25519"
};
function Qt(e) {
  return Xt(e);
}
function Xt(e, t) {
  return e;
}
function No(e) {
  return e;
}
function Ro(e) {
  let t = !0;
  return t = t && "key_type" in e, t;
}
function v(e) {
  return $(e, !1);
}
function $(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.key_type === "EcdsaSecp256k1")
      return pe(e, !0);
    if (e.key_type === "EddsaEd25519")
      return ye(e, !0);
  }
  return {
    key_type: Qt(e.key_type)
  };
}
function O(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_type: e.key_type
    };
}
function Fo(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function ho(e) {
  return pe(e, !1);
}
function pe(e, t) {
  return e == null ? e : {
    ...$(e, t),
    key_hex: e.key_hex
  };
}
function bo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...O(e),
      key_hex: e.key_hex
    };
}
function To(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Jo(e) {
  return Yt(e);
}
function Yt(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex
  };
}
function Io(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex
    };
}
function wo(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function vo(e) {
  return ye(e, !1);
}
function ye(e, t) {
  return e == null ? e : {
    ...$(e, t),
    key_hex: e.key_hex
  };
}
function Eo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...O(e),
      key_hex: e.key_hex
    };
}
function Do(e) {
  let t = !0;
  return t = t && "key_hex" in e, t;
}
function Ao(e) {
  return Zt(e);
}
function Zt(e, t) {
  return e == null ? e : {
    key_hex: e.key_hex
  };
}
function qo(e) {
  if (e !== void 0)
    return e === null ? null : {
      key_hex: e.key_hex
    };
}
function Po(e) {
  return !0;
}
function xo(e) {
  return jt(e);
}
function jt(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0
  };
}
function Co(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor
    };
}
function Vo(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "raw_json" in e, t;
}
function en(e) {
  return tn(e);
}
function tn(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    raw_json: e.raw_json
  };
}
function nn(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      raw_json: e.raw_json
    };
}
function ko(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function Mo(e) {
  return rn(e);
}
function rn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    addresses: e.addresses,
    aggregation_level: n(e, "aggregation_level") ? p(e.aggregation_level) : void 0
  };
}
function sn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      addresses: e.addresses,
      aggregation_level: e.aggregation_level
    };
}
function Lo(e) {
  let t = !0;
  return t = t && "addresses" in e, t;
}
function Ko(e) {
  return on(e);
}
function on(e, t) {
  return e == null ? e : {
    addresses: e.addresses,
    aggregation_level: n(e, "aggregation_level") ? p(e.aggregation_level) : void 0
  };
}
function Go(e) {
  if (e !== void 0)
    return e === null ? null : {
      addresses: e.addresses,
      aggregation_level: e.aggregation_level
    };
}
function Uo(e) {
  return !0;
}
function an(e) {
  return un(e);
}
function un(e, t) {
  return e == null ? e : {
    parent_address: n(e, "parent_address") ? e.parent_address : void 0,
    owner_address: n(e, "owner_address") ? e.owner_address : void 0,
    global_address: n(e, "global_address") ? e.global_address : void 0
  };
}
function ln(e) {
  if (e !== void 0)
    return e === null ? null : {
      parent_address: e.parent_address,
      owner_address: e.owner_address,
      global_address: e.global_address
    };
}
const $o = {
  FungibleResource: "FungibleResource",
  NonFungibleResource: "NonFungibleResource",
  FungibleVault: "FungibleVault",
  NonFungibleVault: "NonFungibleVault",
  Package: "Package",
  Component: "Component"
};
function cn(e) {
  return dn(e);
}
function dn(e, t) {
  return e;
}
function Bo(e) {
  return e;
}
function zo(e) {
  let t = !0;
  return t = t && "type" in e, t;
}
function _n(e) {
  return F(e, !1);
}
function F(e, t) {
  if (e == null)
    return e;
  if (!t) {
    if (e.type === "Component")
      return Ne(e, !0);
    if (e.type === "FungibleResource")
      return Re(e, !0);
    if (e.type === "FungibleVault")
      return Fe(e);
    if (e.type === "NonFungibleResource")
      return he(e, !0);
    if (e.type === "NonFungibleVault")
      return be(e);
    if (e.type === "Package")
      return Te(e, !0);
  }
  return {
    type: cn(e.type)
  };
}
function h(e) {
  if (e !== void 0)
    return e === null ? null : {
      type: e.type
    };
}
function Wo(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "metadata" in e, t;
}
function Se(e) {
  return fn(e);
}
function fn(e, t) {
  return e == null ? e : {
    address: e.address,
    fungible_resources: n(e, "fungible_resources") ? dt(e.fungible_resources) : void 0,
    non_fungible_resources: n(e, "non_fungible_resources") ? Mt(e.non_fungible_resources) : void 0,
    ancestor_identities: n(e, "ancestor_identities") ? an(e.ancestor_identities) : void 0,
    metadata: H(e.metadata),
    details: n(e, "details") ? _n(e.details) : void 0
  };
}
function Oe(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      fungible_resources: ft(e.fungible_resources),
      non_fungible_resources: Kt(e.non_fungible_resources),
      ancestor_identities: ln(e.ancestor_identities),
      metadata: Q(e.metadata),
      details: h(e.details)
    };
}
function Ho(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t;
}
function gn(e) {
  return mn(e);
}
function mn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    items: e.items.map(Se)
  };
}
function Qo(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      items: e.items.map(Oe)
    };
}
function Xo(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Yo(e) {
  return pn(e);
}
function pn(e, t) {
  return e == null ? e : {
    items: e.items.map(Se)
  };
}
function Zo(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(Oe)
    };
}
function jo(e) {
  let t = !0;
  return t = t && "blueprint_name" in e, t = t && "access_rules_chain" in e, t;
}
function ea(e) {
  return Ne(e, !1);
}
function Ne(e, t) {
  return e == null ? e : {
    ...F(e, t),
    package_address: n(e, "package_address") ? e.package_address : void 0,
    blueprint_name: e.blueprint_name,
    state: n(e, "state") ? e.state : void 0,
    access_rules_chain: e.access_rules_chain,
    royalty_aggregator: n(e, "royalty_aggregator") ? J(e.royalty_aggregator) : void 0
  };
}
function ta(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...h(e),
      package_address: e.package_address,
      blueprint_name: e.blueprint_name,
      state: e.state,
      access_rules_chain: e.access_rules_chain,
      royalty_aggregator: I(e.royalty_aggregator)
    };
}
function na(e) {
  let t = !0;
  return t = t && "blueprint_name" in e, t = t && "access_rules_chain" in e, t;
}
function ra(e) {
  return yn(e);
}
function yn(e, t) {
  return e == null ? e : {
    package_address: n(e, "package_address") ? e.package_address : void 0,
    blueprint_name: e.blueprint_name,
    state: n(e, "state") ? e.state : void 0,
    access_rules_chain: e.access_rules_chain,
    royalty_aggregator: n(e, "royalty_aggregator") ? J(e.royalty_aggregator) : void 0
  };
}
function ia(e) {
  if (e !== void 0)
    return e === null ? null : {
      package_address: e.package_address,
      blueprint_name: e.blueprint_name,
      state: e.state,
      access_rules_chain: e.access_rules_chain,
      royalty_aggregator: I(e.royalty_aggregator)
    };
}
function sa(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "divisibility" in e, t;
}
function oa(e) {
  return Re(e, !1);
}
function Re(e, t) {
  return e == null ? e : {
    ...F(e, t),
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    divisibility: e.divisibility
  };
}
function aa(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...h(e),
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      divisibility: e.divisibility
    };
}
function ua(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "divisibility" in e, t;
}
function la(e) {
  return Sn(e);
}
function Sn(e, t) {
  return e == null ? e : {
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    divisibility: e.divisibility
  };
}
function ca(e) {
  if (e !== void 0)
    return e === null ? null : {
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      divisibility: e.divisibility
    };
}
function da(e) {
  return !0;
}
function _a(e) {
  return Fe(e);
}
function Fe(e, t) {
  return e;
}
function fa(e) {
  return e;
}
function ga(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "non_fungible_id_type" in e, t;
}
function ma(e) {
  return he(e, !1);
}
function he(e, t) {
  return e == null ? e : {
    ...F(e, t),
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    non_fungible_id_type: w(e.non_fungible_id_type)
  };
}
function pa(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...h(e),
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      non_fungible_id_type: e.non_fungible_id_type
    };
}
function ya(e) {
  let t = !0;
  return t = t && "access_rules_chain" in e, t = t && "vault_access_rules_chain" in e, t = t && "non_fungible_id_type" in e, t;
}
function Sa(e) {
  return On(e);
}
function On(e, t) {
  return e == null ? e : {
    access_rules_chain: e.access_rules_chain,
    vault_access_rules_chain: e.vault_access_rules_chain,
    non_fungible_id_type: w(e.non_fungible_id_type)
  };
}
function Oa(e) {
  if (e !== void 0)
    return e === null ? null : {
      access_rules_chain: e.access_rules_chain,
      vault_access_rules_chain: e.vault_access_rules_chain,
      non_fungible_id_type: e.non_fungible_id_type
    };
}
function Na(e) {
  return !0;
}
function Ra(e) {
  return be(e);
}
function be(e, t) {
  return e;
}
function Fa(e) {
  return e;
}
function ha(e) {
  return !0;
}
function ba(e) {
  return Te(e, !1);
}
function Te(e, t) {
  return e == null ? e : {
    ...F(e, t),
    code_hex: n(e, "code_hex") ? e.code_hex : void 0,
    royalty_aggregator: n(e, "royalty_aggregator") ? J(e.royalty_aggregator) : void 0
  };
}
function Ta(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...h(e),
      code_hex: e.code_hex,
      royalty_aggregator: I(e.royalty_aggregator)
    };
}
function Ja(e) {
  return !0;
}
function Ia(e) {
  return Nn(e);
}
function Nn(e, t) {
  return e == null ? e : {
    code_hex: n(e, "code_hex") ? e.code_hex : void 0,
    royalty_aggregator: n(e, "royalty_aggregator") ? J(e.royalty_aggregator) : void 0
  };
}
function wa(e) {
  if (e !== void 0)
    return e === null ? null : {
      code_hex: e.code_hex,
      royalty_aggregator: I(e.royalty_aggregator)
    };
}
function va(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Ea(e) {
  return Rn(e);
}
function Rn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    resource_address: e.resource_address
  };
}
function Fn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      resource_address: e.resource_address
    };
}
function Da(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Aa(e) {
  return hn(e);
}
function hn(e, t) {
  return e == null ? e : {
    address: e.address,
    resource_address: e.resource_address
  };
}
function qa(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      resource_address: e.resource_address
    };
}
function Pa(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function bn(e) {
  return Tn(e);
}
function Tn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(C),
    address: e.address,
    resource_address: e.resource_address
  };
}
function xa(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(V),
      address: e.address,
      resource_address: e.resource_address
    };
}
function Ca(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function Va(e) {
  return Jn(e);
}
function Jn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    aggregation_level: n(e, "aggregation_level") ? p(e.aggregation_level) : void 0
  };
}
function In(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function ka(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function Ma(e) {
  return wn(e);
}
function wn(e, t) {
  return e == null ? e : {
    address: e.address,
    aggregation_level: n(e, "aggregation_level") ? p(e.aggregation_level) : void 0
  };
}
function La(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function Ka(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function vn(e) {
  return En(e);
}
function En(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(P),
    address: e.address
  };
}
function Ga(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(N),
      address: e.address
    };
}
function Ua(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function $a(e) {
  return Dn(e);
}
function Dn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address
  };
}
function An(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address
    };
}
function Ba(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function za(e) {
  return qn(e);
}
function qn(e, t) {
  return e == null ? e : {
    address: e.address
  };
}
function Wa(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address
    };
}
function Ha(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function Pn(e) {
  return xn(e);
}
function xn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(A),
    address: e.address
  };
}
function Qa(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(q),
      address: e.address
    };
}
function Xa(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "vault_address" in e, t = t && "resource_address" in e, t;
}
function Ya(e) {
  return Cn(e);
}
function Cn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    vault_address: e.vault_address,
    resource_address: e.resource_address
  };
}
function Vn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      vault_address: e.vault_address,
      resource_address: e.resource_address
    };
}
function Za(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "vault_address" in e, t = t && "resource_address" in e, t;
}
function ja(e) {
  return kn(e);
}
function kn(e, t) {
  return e == null ? e : {
    address: e.address,
    vault_address: e.vault_address,
    resource_address: e.resource_address
  };
}
function eu(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      vault_address: e.vault_address,
      resource_address: e.resource_address
    };
}
function tu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Mn(e) {
  return Ln(e);
}
function Ln(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(k),
    address: e.address,
    resource_address: e.resource_address
  };
}
function nu(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(M),
      address: e.address,
      resource_address: e.resource_address
    };
}
function ru(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "resource_address" in e, t;
}
function iu(e) {
  return Kn(e);
}
function Kn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    resource_address: e.resource_address
  };
}
function Gn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      resource_address: e.resource_address
    };
}
function su(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t = t && "resource_address" in e, t;
}
function Un(e) {
  return $n(e);
}
function $n(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(G),
    address: e.address,
    resource_address: e.resource_address
  };
}
function ou(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(U),
      address: e.address,
      resource_address: e.resource_address
    };
}
function au(e) {
  let t = !0;
  return t = t && "address" in e, t;
}
function uu(e) {
  return Bn(e);
}
function Bn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    address: e.address,
    aggregation_level: n(e, "aggregation_level") ? p(e.aggregation_level) : void 0
  };
}
function zn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      address: e.address,
      aggregation_level: e.aggregation_level
    };
}
function lu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t = t && "address" in e, t;
}
function Wn(e) {
  return Hn(e);
}
function Hn(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(L),
    address: e.address
  };
}
function cu(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(R),
      address: e.address
    };
}
function du(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function _u(e) {
  return Qn(e);
}
function Qn(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    resource_address: e.resource_address,
    non_fungible_ids: e.non_fungible_ids
  };
}
function Xn(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      resource_address: e.resource_address,
      non_fungible_ids: e.non_fungible_ids
    };
}
function fu(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function gu(e) {
  return Yn(e);
}
function Yn(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_ids: e.non_fungible_ids
  };
}
function mu(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_ids: e.non_fungible_ids
    };
}
function pu(e) {
  let t = !0;
  return t = t && "non_fungible_id" in e, t = t && "mutable_data" in e, t = t && "last_updated_at_state_version" in e, t;
}
function Je(e) {
  return Zn(e);
}
function Zn(e, t) {
  return e == null ? e : {
    non_fungible_id: e.non_fungible_id,
    mutable_data: en(e.mutable_data),
    last_updated_at_state_version: e.last_updated_at_state_version
  };
}
function Ie(e) {
  if (e !== void 0)
    return e === null ? null : {
      non_fungible_id: e.non_fungible_id,
      mutable_data: nn(e.mutable_data),
      last_updated_at_state_version: e.last_updated_at_state_version
    };
}
function yu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "resource_address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_ids" in e, t;
}
function jn(e) {
  return er(e);
}
function er(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    resource_address: e.resource_address,
    non_fungible_id_type: w(e.non_fungible_id_type),
    non_fungible_ids: e.non_fungible_ids.map(Je)
  };
}
function Su(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      resource_address: e.resource_address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_ids: e.non_fungible_ids.map(Ie)
    };
}
function Ou(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_id_type" in e, t = t && "non_fungible_ids" in e, t;
}
function Nu(e) {
  return tr(e);
}
function tr(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_id_type: w(e.non_fungible_id_type),
    non_fungible_ids: e.non_fungible_ids.map(Je)
  };
}
function Ru(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_id_type: e.non_fungible_id_type,
      non_fungible_ids: e.non_fungible_ids.map(Ie)
    };
}
function Fu(e) {
  let t = !0;
  return t = t && "resource_address" in e, t;
}
function hu(e) {
  return nr(e);
}
function nr(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    resource_address: e.resource_address
  };
}
function rr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      resource_address: e.resource_address
    };
}
function bu(e) {
  let t = !0;
  return t = t && "resource_address" in e, t;
}
function Tu(e) {
  return ir(e);
}
function ir(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address
  };
}
function Ju(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address
    };
}
function Iu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function sr(e) {
  return or(e);
}
function or(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    resource_address: e.resource_address,
    non_fungible_ids: le(e.non_fungible_ids)
  };
}
function wu(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      resource_address: e.resource_address,
      non_fungible_ids: ce(e.non_fungible_ids)
    };
}
function vu(e) {
  let t = !0;
  return t = t && "resource_address" in e, t = t && "non_fungible_ids" in e, t;
}
function Eu(e) {
  return ar(e);
}
function ar(e, t) {
  return e == null ? e : {
    resource_address: e.resource_address,
    non_fungible_ids: le(e.non_fungible_ids)
  };
}
function Du(e) {
  if (e !== void 0)
    return e === null ? null : {
      resource_address: e.resource_address,
      non_fungible_ids: ce(e.non_fungible_ids)
    };
}
function Au(e) {
  return !0;
}
function qu(e) {
  return ur(e);
}
function ur(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0
  };
}
function lr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor
    };
}
function Pu(e) {
  return !0;
}
function xu(e) {
  return cr(e);
}
function cr(e, t) {
  return e == null ? e : {
    cursor: n(e, "cursor") ? e.cursor : void 0
  };
}
function Cu(e) {
  if (e !== void 0)
    return e === null ? null : {
      cursor: e.cursor
    };
}
function Vu(e) {
  let t = !0;
  return t = t && "stake" in e, t = t && "stake_percentage" in e, t = t && "key" in e, t;
}
function dr(e) {
  return _r(e);
}
function _r(e, t) {
  return e == null ? e : {
    stake: e.stake,
    stake_percentage: e.stake_percentage,
    key: v(e.key)
  };
}
function fr(e) {
  if (e !== void 0)
    return e === null ? null : {
      stake: e.stake,
      stake_percentage: e.stake_percentage,
      key: O(e.key)
    };
}
function ku(e) {
  let t = !0;
  return t = t && "address" in e, t = t && "current_stake" in e, t = t && "metadata" in e, t;
}
function we(e) {
  return gr(e);
}
function gr(e, t) {
  return e == null ? e : {
    address: e.address,
    state: n(e, "state") ? e.state : void 0,
    current_stake: e.current_stake,
    active_in_epoch: n(e, "active_in_epoch") ? dr(e.active_in_epoch) : void 0,
    metadata: H(e.metadata)
  };
}
function ve(e) {
  if (e !== void 0)
    return e === null ? null : {
      address: e.address,
      state: e.state,
      current_stake: e.current_stake,
      active_in_epoch: fr(e.active_in_epoch),
      metadata: Q(e.metadata)
    };
}
function Mu(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Ee(e) {
  return mr(e);
}
function mr(e, t) {
  return e == null ? e : {
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(we)
  };
}
function De(e) {
  if (e !== void 0)
    return e === null ? null : {
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(ve)
    };
}
function Lu(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "validators" in e, t;
}
function pr(e) {
  return yr(e);
}
function yr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    validators: Ee(e.validators)
  };
}
function Ku(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      validators: De(e.validators)
    };
}
function Gu(e) {
  let t = !0;
  return t = t && "validators" in e, t;
}
function Uu(e) {
  return Sr(e);
}
function Sr(e, t) {
  return e == null ? e : {
    validators: Ee(e.validators)
  };
}
function $u(e) {
  if (e !== void 0)
    return e === null ? null : {
      validators: De(e.validators)
    };
}
const Bu = {
  User: "User",
  EpochChange: "EpochChange",
  All: "All"
}, zu = {
  Asc: "Asc",
  Desc: "Desc"
};
function Wu(e) {
  return !0;
}
function Hu(e) {
  return Or(e);
}
function Or(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    cursor: n(e, "cursor") ? e.cursor : void 0,
    limit_per_page: n(e, "limit_per_page") ? e.limit_per_page : void 0,
    from_ledger_state: n(e, "from_ledger_state") ? _(e.from_ledger_state) : void 0,
    kind_filter: n(e, "kind_filter") ? e.kind_filter : void 0,
    order: n(e, "order") ? e.order : void 0
  };
}
function Nr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      cursor: e.cursor,
      limit_per_page: e.limit_per_page,
      from_ledger_state: f(e.from_ledger_state),
      kind_filter: e.kind_filter,
      order: e.order
    };
}
const Qu = {
  User: "User",
  EpochChange: "EpochChange",
  All: "All"
}, Xu = {
  Asc: "Asc",
  Desc: "Desc"
};
function Yu(e) {
  return !0;
}
function Zu(e) {
  return Rr(e);
}
function Rr(e, t) {
  return e == null ? e : {
    from_ledger_state: n(e, "from_ledger_state") ? _(e.from_ledger_state) : void 0,
    kind_filter: n(e, "kind_filter") ? e.kind_filter : void 0,
    order: n(e, "order") ? e.order : void 0
  };
}
function ju(e) {
  if (e !== void 0)
    return e === null ? null : {
      from_ledger_state: f(e.from_ledger_state),
      kind_filter: e.kind_filter,
      order: e.order
    };
}
function el(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "items" in e, t;
}
function Fr(e) {
  return hr(e);
}
function hr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    total_count: n(e, "total_count") ? e.total_count : void 0,
    previous_cursor: n(e, "previous_cursor") ? e.previous_cursor : void 0,
    next_cursor: n(e, "next_cursor") ? e.next_cursor : void 0,
    items: e.items.map(E)
  };
}
function tl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      total_count: e.total_count,
      previous_cursor: e.previous_cursor,
      next_cursor: e.next_cursor,
      items: e.items.map(D)
    };
}
function nl(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function rl(e) {
  return br(e);
}
function br(e, t) {
  return e == null ? e : {
    at_ledger_state: n(e, "at_ledger_state") ? _(e.at_ledger_state) : void 0,
    intent_hash_hex: e.intent_hash_hex
  };
}
function Tr(e) {
  if (e !== void 0)
    return e === null ? null : {
      at_ledger_state: f(e.at_ledger_state),
      intent_hash_hex: e.intent_hash_hex
    };
}
function il(e) {
  let t = !0;
  return t = t && "raw_hex" in e, t = t && "receipt" in e, t = t && "referenced_global_entities" in e, t;
}
function Ae(e) {
  return Jr(e);
}
function Jr(e, t) {
  return e == null ? e : {
    raw_hex: e.raw_hex,
    receipt: e.receipt,
    referenced_global_entities: e.referenced_global_entities,
    message_hex: n(e, "message_hex") ? e.message_hex : void 0
  };
}
function qe(e) {
  if (e !== void 0)
    return e === null ? null : {
      raw_hex: e.raw_hex,
      receipt: e.receipt,
      referenced_global_entities: e.referenced_global_entities,
      message_hex: e.message_hex
    };
}
function sl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "transaction" in e, t = t && "details" in e, t;
}
function Ir(e) {
  return wr(e);
}
function wr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    transaction: E(e.transaction),
    details: Ae(e.details)
  };
}
function ol(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      transaction: D(e.transaction),
      details: qe(e.details)
    };
}
function al(e) {
  let t = !0;
  return t = t && "transaction" in e, t = t && "details" in e, t;
}
function ul(e) {
  return vr(e);
}
function vr(e, t) {
  return e == null ? e : {
    transaction: E(e.transaction),
    details: Ae(e.details)
  };
}
function ll(e) {
  if (e !== void 0)
    return e === null ? null : {
      transaction: D(e.transaction),
      details: qe(e.details)
    };
}
function cl(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t;
}
function Er(e) {
  return Dr(e);
}
function Dr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state)
  };
}
function dl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state)
    };
}
function _l(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function fl(e) {
  return Pe(e, !1);
}
function Pe(e, t) {
  return e == null ? e : {
    ...g(e, t),
    intent_hash_hex: e.intent_hash_hex
  };
}
function gl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ...m(e),
      intent_hash_hex: e.intent_hash_hex
    };
}
function ml(e) {
  let t = !0;
  return t = t && "unlimited_loan" in e, t = t && "assume_all_signature_proofs" in e, t = t && "permit_duplicate_intent_hash" in e, t = t && "permit_invalid_header_epoch" in e, t;
}
function Ar(e) {
  return qr(e);
}
function qr(e, t) {
  return e == null ? e : {
    unlimited_loan: e.unlimited_loan,
    assume_all_signature_proofs: e.assume_all_signature_proofs,
    permit_duplicate_intent_hash: e.permit_duplicate_intent_hash,
    permit_invalid_header_epoch: e.permit_invalid_header_epoch
  };
}
function Pr(e) {
  if (e !== void 0)
    return e === null ? null : {
      unlimited_loan: e.unlimited_loan,
      assume_all_signature_proofs: e.assume_all_signature_proofs,
      permit_duplicate_intent_hash: e.permit_duplicate_intent_hash,
      permit_invalid_header_epoch: e.permit_invalid_header_epoch
    };
}
function pl(e) {
  let t = !0;
  return t = t && "manifest" in e, t = t && "start_epoch_inclusive" in e, t = t && "end_epoch_exclusive" in e, t = t && "cost_unit_limit" in e, t = t && "tip_percentage" in e, t = t && "nonce" in e, t = t && "signer_public_keys" in e, t = t && "flags" in e, t;
}
function yl(e) {
  return xr(e);
}
function xr(e, t) {
  return e == null ? e : {
    manifest: e.manifest,
    blobs_hex: n(e, "blobs_hex") ? e.blobs_hex : void 0,
    start_epoch_inclusive: e.start_epoch_inclusive,
    end_epoch_exclusive: e.end_epoch_exclusive,
    notary_public_key: n(e, "notary_public_key") ? v(e.notary_public_key) : void 0,
    notary_as_signatory: n(e, "notary_as_signatory") ? e.notary_as_signatory : void 0,
    cost_unit_limit: e.cost_unit_limit,
    tip_percentage: e.tip_percentage,
    nonce: e.nonce,
    signer_public_keys: e.signer_public_keys.map(v),
    flags: Ar(e.flags)
  };
}
function Cr(e) {
  if (e !== void 0)
    return e === null ? null : {
      manifest: e.manifest,
      blobs_hex: e.blobs_hex,
      start_epoch_inclusive: e.start_epoch_inclusive,
      end_epoch_exclusive: e.end_epoch_exclusive,
      notary_public_key: O(e.notary_public_key),
      notary_as_signatory: e.notary_as_signatory,
      cost_unit_limit: e.cost_unit_limit,
      tip_percentage: e.tip_percentage,
      nonce: e.nonce,
      signer_public_keys: e.signer_public_keys.map(O),
      flags: Pr(e.flags)
    };
}
function Sl(e) {
  let t = !0;
  return t = t && "level" in e, t = t && "message" in e, t;
}
function Vr(e) {
  return kr(e);
}
function kr(e, t) {
  return e == null ? e : {
    level: e.level,
    message: e.message
  };
}
function Mr(e) {
  if (e !== void 0)
    return e === null ? null : {
      level: e.level,
      message: e.message
    };
}
function Ol(e) {
  let t = !0;
  return t = t && "encoded_receipt" in e, t = t && "receipt" in e, t = t && "resource_changes" in e, t = t && "logs" in e, t;
}
function Lr(e) {
  return Kr(e);
}
function Kr(e, t) {
  return e == null ? e : {
    encoded_receipt: e.encoded_receipt,
    receipt: e.receipt,
    resource_changes: e.resource_changes,
    logs: e.logs.map(Vr)
  };
}
function Nl(e) {
  if (e !== void 0)
    return e === null ? null : {
      encoded_receipt: e.encoded_receipt,
      receipt: e.receipt,
      resource_changes: e.resource_changes,
      logs: e.logs.map(Mr)
    };
}
function Rl(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function Fl(e) {
  return Gr(e);
}
function Gr(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function Ur(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function hl(e) {
  let t = !0;
  return t = t && "intent_hash_hex" in e, t;
}
function bl(e) {
  return $r(e);
}
function $r(e, t) {
  return e == null ? e : {
    intent_hash_hex: e.intent_hash_hex
  };
}
function Tl(e) {
  if (e !== void 0)
    return e === null ? null : {
      intent_hash_hex: e.intent_hash_hex
    };
}
function Jl(e) {
  let t = !0;
  return t = t && "payload_hash_hex" in e, t = t && "status" in e, t;
}
function xe(e) {
  return Br(e);
}
function Br(e, t) {
  return e == null ? e : {
    payload_hash_hex: e.payload_hash_hex,
    status: T(e.status),
    error_message: n(e, "error_message") ? e.error_message : void 0
  };
}
function Ce(e) {
  if (e !== void 0)
    return e === null ? null : {
      payload_hash_hex: e.payload_hash_hex,
      status: e.status,
      error_message: e.error_message
    };
}
function Il(e) {
  let t = !0;
  return t = t && "ledger_state" in e, t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function zr(e) {
  return Wr(e);
}
function Wr(e, t) {
  return e == null ? e : {
    ledger_state: l(e.ledger_state),
    status: T(e.status),
    known_payloads: e.known_payloads.map(xe),
    error_message: n(e, "error_message") ? e.error_message : void 0
  };
}
function wl(e) {
  if (e !== void 0)
    return e === null ? null : {
      ledger_state: c(e.ledger_state),
      status: e.status,
      known_payloads: e.known_payloads.map(Ce),
      error_message: e.error_message
    };
}
function vl(e) {
  let t = !0;
  return t = t && "status" in e, t = t && "known_payloads" in e, t;
}
function El(e) {
  return Hr(e);
}
function Hr(e, t) {
  return e == null ? e : {
    status: T(e.status),
    known_payloads: e.known_payloads.map(xe),
    error_message: n(e, "error_message") ? e.error_message : void 0
  };
}
function Dl(e) {
  if (e !== void 0)
    return e === null ? null : {
      status: e.status,
      known_payloads: e.known_payloads.map(Ce),
      error_message: e.error_message
    };
}
function Al(e) {
  let t = !0;
  return t = t && "notarized_transaction_hex" in e, t;
}
function ql(e) {
  return Qr(e);
}
function Qr(e, t) {
  return e == null ? e : {
    notarized_transaction_hex: e.notarized_transaction_hex
  };
}
function Xr(e) {
  if (e !== void 0)
    return e === null ? null : {
      notarized_transaction_hex: e.notarized_transaction_hex
    };
}
function Pl(e) {
  let t = !0;
  return t = t && "duplicate" in e, t;
}
function Yr(e) {
  return Zr(e);
}
function Zr(e, t) {
  return e == null ? e : {
    duplicate: e.duplicate
  };
}
function xl(e) {
  if (e !== void 0)
    return e === null ? null : {
      duplicate: e.duplicate
    };
}
function Cl(e) {
  let t = !0;
  return t = t && "items" in e, t;
}
function Vl(e) {
  return jr(e);
}
function jr(e, t) {
  return e == null ? e : {
    items: e.items.map(we)
  };
}
function kl(e) {
  if (e !== void 0)
    return e === null ? null : {
      items: e.items.map(ve)
    };
}
class Ml extends b {
  async entityFungibleResourceVaultPageRaw(t, r) {
    if (t.stateEntityFungibleResourceVaultsPageRequest === null || t.stateEntityFungibleResourceVaultsPageRequest === void 0)
      throw new d("stateEntityFungibleResourceVaultsPageRequest", "Required parameter requestParameters.stateEntityFungibleResourceVaultsPageRequest was null or undefined when calling entityFungibleResourceVaultPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/fungible-vaults/",
      method: "POST",
      headers: s,
      query: i,
      body: Fn(t.stateEntityFungibleResourceVaultsPageRequest)
    }, r);
    return new u(o, (a) => bn(a));
  }
  async entityFungibleResourceVaultPage(t, r) {
    return await (await this.entityFungibleResourceVaultPageRaw(t, r)).value();
  }
  async entityFungiblesPageRaw(t, r) {
    if (t.stateEntityFungiblesPageRequest === null || t.stateEntityFungiblesPageRequest === void 0)
      throw new d("stateEntityFungiblesPageRequest", "Required parameter requestParameters.stateEntityFungiblesPageRequest was null or undefined when calling entityFungiblesPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/fungibles/",
      method: "POST",
      headers: s,
      query: i,
      body: In(t.stateEntityFungiblesPageRequest)
    }, r);
    return new u(o, (a) => vn(a));
  }
  async entityFungiblesPage(t, r) {
    return await (await this.entityFungiblesPageRaw(t, r)).value();
  }
  async entityMetadataPageRaw(t, r) {
    if (t.stateEntityMetadataPageRequest === null || t.stateEntityMetadataPageRequest === void 0)
      throw new d("stateEntityMetadataPageRequest", "Required parameter requestParameters.stateEntityMetadataPageRequest was null or undefined when calling entityMetadataPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/metadata",
      method: "POST",
      headers: s,
      query: i,
      body: An(t.stateEntityMetadataPageRequest)
    }, r);
    return new u(o, (a) => Pn(a));
  }
  async entityMetadataPage(t, r) {
    return await (await this.entityMetadataPageRaw(t, r)).value();
  }
  async entityNonFungibleIdsPageRaw(t, r) {
    if (t.stateEntityNonFungibleIdsPageRequest === null || t.stateEntityNonFungibleIdsPageRequest === void 0)
      throw new d("stateEntityNonFungibleIdsPageRequest", "Required parameter requestParameters.stateEntityNonFungibleIdsPageRequest was null or undefined when calling entityNonFungibleIdsPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/non-fungible-vault/ids",
      method: "POST",
      headers: s,
      query: i,
      body: Vn(t.stateEntityNonFungibleIdsPageRequest)
    }, r);
    return new u(o, (a) => Mn(a));
  }
  async entityNonFungibleIdsPage(t, r) {
    return await (await this.entityNonFungibleIdsPageRaw(t, r)).value();
  }
  async entityNonFungibleResourceVaultPageRaw(t, r) {
    if (t.stateEntityNonFungibleResourceVaultsPageRequest === null || t.stateEntityNonFungibleResourceVaultsPageRequest === void 0)
      throw new d("stateEntityNonFungibleResourceVaultsPageRequest", "Required parameter requestParameters.stateEntityNonFungibleResourceVaultsPageRequest was null or undefined when calling entityNonFungibleResourceVaultPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/non-fungible-vaults/",
      method: "POST",
      headers: s,
      query: i,
      body: Gn(t.stateEntityNonFungibleResourceVaultsPageRequest)
    }, r);
    return new u(o, (a) => Un(a));
  }
  async entityNonFungibleResourceVaultPage(t, r) {
    return await (await this.entityNonFungibleResourceVaultPageRaw(t, r)).value();
  }
  async entityNonFungiblesPageRaw(t, r) {
    if (t.stateEntityNonFungiblesPageRequest === null || t.stateEntityNonFungiblesPageRequest === void 0)
      throw new d("stateEntityNonFungiblesPageRequest", "Required parameter requestParameters.stateEntityNonFungiblesPageRequest was null or undefined when calling entityNonFungiblesPage.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/page/non-fungibles/",
      method: "POST",
      headers: s,
      query: i,
      body: zn(t.stateEntityNonFungiblesPageRequest)
    }, r);
    return new u(o, (a) => Wn(a));
  }
  async entityNonFungiblesPage(t, r) {
    return await (await this.entityNonFungiblesPageRaw(t, r)).value();
  }
  async nonFungibleDataRaw(t, r) {
    if (t.stateNonFungibleDataRequest === null || t.stateNonFungibleDataRequest === void 0)
      throw new d("stateNonFungibleDataRequest", "Required parameter requestParameters.stateNonFungibleDataRequest was null or undefined when calling nonFungibleData.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/non-fungible/data",
      method: "POST",
      headers: s,
      query: i,
      body: Xn(t.stateNonFungibleDataRequest)
    }, r);
    return new u(o, (a) => jn(a));
  }
  async nonFungibleData(t, r) {
    return await (await this.nonFungibleDataRaw(t, r)).value();
  }
  async nonFungibleIdsRaw(t, r) {
    if (t.stateNonFungibleIdsRequest === null || t.stateNonFungibleIdsRequest === void 0)
      throw new d("stateNonFungibleIdsRequest", "Required parameter requestParameters.stateNonFungibleIdsRequest was null or undefined when calling nonFungibleIds.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/non-fungible/ids",
      method: "POST",
      headers: s,
      query: i,
      body: rr(t.stateNonFungibleIdsRequest)
    }, r);
    return new u(o, (a) => sr(a));
  }
  async nonFungibleIds(t, r) {
    return await (await this.nonFungibleIdsRaw(t, r)).value();
  }
  async stateEntityDetailsRaw(t, r) {
    if (t.stateEntityDetailsRequest === null || t.stateEntityDetailsRequest === void 0)
      throw new d("stateEntityDetailsRequest", "Required parameter requestParameters.stateEntityDetailsRequest was null or undefined when calling stateEntityDetails.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/entity/details",
      method: "POST",
      headers: s,
      query: i,
      body: sn(t.stateEntityDetailsRequest)
    }, r);
    return new u(o, (a) => gn(a));
  }
  async stateEntityDetails(t, r) {
    return await (await this.stateEntityDetailsRaw(t, r)).value();
  }
  async stateValidatorsListRaw(t, r) {
    if (t.stateValidatorsListRequest === null || t.stateValidatorsListRequest === void 0)
      throw new d("stateValidatorsListRequest", "Required parameter requestParameters.stateValidatorsListRequest was null or undefined when calling stateValidatorsList.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/validators/list",
      method: "POST",
      headers: s,
      query: i,
      body: lr(t.stateValidatorsListRequest)
    }, r);
    return new u(o, (a) => pr(a));
  }
  async stateValidatorsList(t, r) {
    return await (await this.stateValidatorsListRaw(t, r)).value();
  }
}
class Ll extends b {
  async gatewayStatusRaw(t) {
    const r = {}, i = {}, s = await this.request({
      path: "/status/gateway-status",
      method: "POST",
      headers: i,
      query: r
    }, t);
    return new u(s, (o) => ht(o));
  }
  async gatewayStatus(t) {
    return await (await this.gatewayStatusRaw(t)).value();
  }
  async networkConfigurationRaw(t) {
    const r = {}, i = {}, s = await this.request({
      path: "/status/network-configuration",
      method: "POST",
      headers: i,
      query: r
    }, t);
    return new u(s, (o) => qt(o));
  }
  async networkConfiguration(t) {
    return await (await this.networkConfigurationRaw(t)).value();
  }
}
class Kl extends b {
  async streamTransactionsRaw(t, r) {
    if (t.streamTransactionsRequest === null || t.streamTransactionsRequest === void 0)
      throw new d("streamTransactionsRequest", "Required parameter requestParameters.streamTransactionsRequest was null or undefined when calling streamTransactions.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/stream/transactions",
      method: "POST",
      headers: s,
      query: i,
      body: Nr(t.streamTransactionsRequest)
    }, r);
    return new u(o, (a) => Fr(a));
  }
  async streamTransactions(t, r) {
    return await (await this.streamTransactionsRaw(t, r)).value();
  }
}
class Gl extends b {
  async transactionCommittedDetailsRaw(t, r) {
    if (t.transactionCommittedDetailsRequest === null || t.transactionCommittedDetailsRequest === void 0)
      throw new d("transactionCommittedDetailsRequest", "Required parameter requestParameters.transactionCommittedDetailsRequest was null or undefined when calling transactionCommittedDetails.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/committed-details",
      method: "POST",
      headers: s,
      query: i,
      body: Tr(t.transactionCommittedDetailsRequest)
    }, r);
    return new u(o, (a) => Ir(a));
  }
  async transactionCommittedDetails(t, r) {
    return await (await this.transactionCommittedDetailsRaw(t, r)).value();
  }
  async transactionConstructionRaw(t) {
    const r = {}, i = {}, s = await this.request({
      path: "/transaction/construction",
      method: "POST",
      headers: i,
      query: r
    }, t);
    return new u(s, (o) => Er(o));
  }
  async transactionConstruction(t) {
    return await (await this.transactionConstructionRaw(t)).value();
  }
  async transactionPreviewRaw(t, r) {
    if (t.transactionPreviewRequest === null || t.transactionPreviewRequest === void 0)
      throw new d("transactionPreviewRequest", "Required parameter requestParameters.transactionPreviewRequest was null or undefined when calling transactionPreview.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/preview",
      method: "POST",
      headers: s,
      query: i,
      body: Cr(t.transactionPreviewRequest)
    }, r);
    return new u(o, (a) => Lr(a));
  }
  async transactionPreview(t, r) {
    return await (await this.transactionPreviewRaw(t, r)).value();
  }
  async transactionStatusRaw(t, r) {
    if (t.transactionStatusRequest === null || t.transactionStatusRequest === void 0)
      throw new d("transactionStatusRequest", "Required parameter requestParameters.transactionStatusRequest was null or undefined when calling transactionStatus.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/status",
      method: "POST",
      headers: s,
      query: i,
      body: Ur(t.transactionStatusRequest)
    }, r);
    return new u(o, (a) => zr(a));
  }
  async transactionStatus(t, r) {
    return await (await this.transactionStatusRaw(t, r)).value();
  }
  async transactionSubmitRaw(t, r) {
    if (t.transactionSubmitRequest === null || t.transactionSubmitRequest === void 0)
      throw new d("transactionSubmitRequest", "Required parameter requestParameters.transactionSubmitRequest was null or undefined when calling transactionSubmit.");
    const i = {}, s = {};
    s["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/submit",
      method: "POST",
      headers: s,
      query: i,
      body: Xr(t.transactionSubmitRequest)
    }, r);
    return new u(o, (a) => Yr(a));
  }
  async transactionSubmit(t, r) {
    return await (await this.transactionSubmitRaw(t, r)).value();
  }
}
export {
  li as AtLedgerStateMixinFromJSON,
  He as AtLedgerStateMixinFromJSONTyped,
  ci as AtLedgerStateMixinToJSON,
  Le as BASE_PATH,
  b as BaseAPI,
  si as BlobApiResponse,
  ti as COLLECTION_FORMATS,
  E as CommittedTransactionInfoFromJSON,
  je as CommittedTransactionInfoFromJSONTyped,
  D as CommittedTransactionInfoToJSON,
  Ke as Configuration,
  pi as CursorLimitMixinFromJSON,
  et as CursorLimitMixinFromJSONTyped,
  yi as CursorLimitMixinToJSON,
  Ge as DefaultConfig,
  Fi as EntityMetadataCollectionAllOfFromJSON,
  ot as EntityMetadataCollectionAllOfFromJSONTyped,
  hi as EntityMetadataCollectionAllOfToJSON,
  H as EntityMetadataCollectionFromJSON,
  st as EntityMetadataCollectionFromJSONTyped,
  Q as EntityMetadataCollectionToJSON,
  A as EntityMetadataItemFromJSON,
  it as EntityMetadataItemFromJSONTyped,
  q as EntityMetadataItemToJSON,
  Ti as EntityMetadataItemValueAllOfFromJSON,
  at as EntityMetadataItemValueAllOfFromJSONTyped,
  Ji as EntityMetadataItemValueAllOfToJSON,
  tt as EntityMetadataItemValueFromJSON,
  nt as EntityMetadataItemValueFromJSONTyped,
  rt as EntityMetadataItemValueToJSON,
  vi as EntityNotFoundErrorFromJSON,
  X as EntityNotFoundErrorFromJSONTyped,
  Ei as EntityNotFoundErrorToJSON,
  Ai as ErrorResponseFromJSON,
  lt as ErrorResponseFromJSONTyped,
  qi as ErrorResponseToJSON,
  ze as FetchError,
  Mi as FungibleResourcesCollectionAllOfFromJSON,
  gt as FungibleResourcesCollectionAllOfFromJSONTyped,
  Li as FungibleResourcesCollectionAllOfToJSON,
  dt as FungibleResourcesCollectionFromJSON,
  _t as FungibleResourcesCollectionFromJSONTyped,
  P as FungibleResourcesCollectionItemFromJSON,
  x as FungibleResourcesCollectionItemFromJSONTyped,
  Ui as FungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSON,
  mt as FungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSONTyped,
  $i as FungibleResourcesCollectionItemGloballyAggregatedAllOfToJSON,
  J as FungibleResourcesCollectionItemGloballyAggregatedFromJSON,
  Y as FungibleResourcesCollectionItemGloballyAggregatedFromJSONTyped,
  I as FungibleResourcesCollectionItemGloballyAggregatedToJSON,
  N as FungibleResourcesCollectionItemToJSON,
  Yi as FungibleResourcesCollectionItemVaultAggregatedAllOfFromJSON,
  St as FungibleResourcesCollectionItemVaultAggregatedAllOfFromJSONTyped,
  Zi as FungibleResourcesCollectionItemVaultAggregatedAllOfToJSON,
  Hi as FungibleResourcesCollectionItemVaultAggregatedFromJSON,
  ee as FungibleResourcesCollectionItemVaultAggregatedFromJSONTyped,
  Qi as FungibleResourcesCollectionItemVaultAggregatedToJSON,
  es as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSON,
  Ot as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSONTyped,
  ts as FungibleResourcesCollectionItemVaultAggregatedVaultAllOfToJSON,
  Z as FungibleResourcesCollectionItemVaultAggregatedVaultFromJSON,
  yt as FungibleResourcesCollectionItemVaultAggregatedVaultFromJSONTyped,
  C as FungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSON,
  pt as FungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSONTyped,
  V as FungibleResourcesCollectionItemVaultAggregatedVaultItemToJSON,
  j as FungibleResourcesCollectionItemVaultAggregatedVaultToJSON,
  ft as FungibleResourcesCollectionToJSON,
  ut as GatewayErrorFromJSON,
  g as GatewayErrorFromJSONTyped,
  m as GatewayErrorToJSON,
  rs as GatewayInfoResponseKnownTargetFromJSON,
  Nt as GatewayInfoResponseKnownTargetFromJSONTyped,
  is as GatewayInfoResponseKnownTargetToJSON,
  te as GatewayInfoResponseReleaseInfoFromJSON,
  Rt as GatewayInfoResponseReleaseInfoFromJSONTyped,
  ne as GatewayInfoResponseReleaseInfoToJSON,
  cs as GatewayStatusResponseAllOfFromJSON,
  Tt as GatewayStatusResponseAllOfFromJSONTyped,
  ds as GatewayStatusResponseAllOfToJSON,
  ht as GatewayStatusResponseFromJSON,
  bt as GatewayStatusResponseFromJSONTyped,
  us as GatewayStatusResponseToJSON,
  ps as InternalServerErrorAllOfFromJSON,
  Jt as InternalServerErrorAllOfFromJSONTyped,
  ys as InternalServerErrorAllOfToJSON,
  fs as InternalServerErrorFromJSON,
  re as InternalServerErrorFromJSONTyped,
  gs as InternalServerErrorToJSON,
  Os as InvalidEntityErrorFromJSON,
  ie as InvalidEntityErrorFromJSONTyped,
  Ns as InvalidEntityErrorToJSON,
  Js as InvalidRequestErrorAllOfFromJSON,
  wt as InvalidRequestErrorAllOfFromJSONTyped,
  Is as InvalidRequestErrorAllOfToJSON,
  hs as InvalidRequestErrorFromJSON,
  ae as InvalidRequestErrorFromJSONTyped,
  bs as InvalidRequestErrorToJSON,
  vs as InvalidTransactionErrorFromJSON,
  ue as InvalidTransactionErrorFromJSONTyped,
  Es as InvalidTransactionErrorToJSON,
  u as JSONApiResponse,
  l as LedgerStateFromJSON,
  Ft as LedgerStateFromJSONTyped,
  As as LedgerStateMixinFromJSON,
  vt as LedgerStateMixinFromJSONTyped,
  qs as LedgerStateMixinToJSON,
  _ as LedgerStateSelectorFromJSON,
  We as LedgerStateSelectorFromJSONTyped,
  f as LedgerStateSelectorToJSON,
  c as LedgerStateToJSON,
  qt as NetworkConfigurationResponseFromJSON,
  Pt as NetworkConfigurationResponseFromJSONTyped,
  Cs as NetworkConfigurationResponseToJSON,
  Et as NetworkConfigurationResponseWellKnownAddressesFromJSON,
  Dt as NetworkConfigurationResponseWellKnownAddressesFromJSONTyped,
  At as NetworkConfigurationResponseWellKnownAddressesToJSON,
  Vs as NonFungibleIdType,
  w as NonFungibleIdTypeFromJSON,
  xt as NonFungibleIdTypeFromJSONTyped,
  ks as NonFungibleIdTypeToJSON,
  Gs as NonFungibleIdsCollectionAllOfFromJSON,
  kt as NonFungibleIdsCollectionAllOfFromJSONTyped,
  Us as NonFungibleIdsCollectionAllOfToJSON,
  le as NonFungibleIdsCollectionFromJSON,
  Vt as NonFungibleIdsCollectionFromJSONTyped,
  k as NonFungibleIdsCollectionItemFromJSON,
  Ct as NonFungibleIdsCollectionItemFromJSONTyped,
  M as NonFungibleIdsCollectionItemToJSON,
  ce as NonFungibleIdsCollectionToJSON,
  Ws as NonFungibleResourcesCollectionAllOfFromJSON,
  Gt as NonFungibleResourcesCollectionAllOfFromJSONTyped,
  Hs as NonFungibleResourcesCollectionAllOfToJSON,
  Mt as NonFungibleResourcesCollectionFromJSON,
  Lt as NonFungibleResourcesCollectionFromJSONTyped,
  L as NonFungibleResourcesCollectionItemFromJSON,
  K as NonFungibleResourcesCollectionItemFromJSONTyped,
  js as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSON,
  Ut as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfFromJSONTyped,
  eo as NonFungibleResourcesCollectionItemGloballyAggregatedAllOfToJSON,
  Xs as NonFungibleResourcesCollectionItemGloballyAggregatedFromJSON,
  de as NonFungibleResourcesCollectionItemGloballyAggregatedFromJSONTyped,
  Ys as NonFungibleResourcesCollectionItemGloballyAggregatedToJSON,
  R as NonFungibleResourcesCollectionItemToJSON,
  ao as NonFungibleResourcesCollectionItemVaultAggregatedAllOfFromJSON,
  zt as NonFungibleResourcesCollectionItemVaultAggregatedAllOfFromJSONTyped,
  uo as NonFungibleResourcesCollectionItemVaultAggregatedAllOfToJSON,
  io as NonFungibleResourcesCollectionItemVaultAggregatedFromJSON,
  ge as NonFungibleResourcesCollectionItemVaultAggregatedFromJSONTyped,
  so as NonFungibleResourcesCollectionItemVaultAggregatedToJSON,
  co as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSON,
  Wt as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfFromJSONTyped,
  _o as NonFungibleResourcesCollectionItemVaultAggregatedVaultAllOfToJSON,
  _e as NonFungibleResourcesCollectionItemVaultAggregatedVaultFromJSON,
  Bt as NonFungibleResourcesCollectionItemVaultAggregatedVaultFromJSONTyped,
  G as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSON,
  $t as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemFromJSONTyped,
  U as NonFungibleResourcesCollectionItemVaultAggregatedVaultItemToJSON,
  fe as NonFungibleResourcesCollectionItemVaultAggregatedVaultToJSON,
  Kt as NonFungibleResourcesCollectionToJSON,
  yo as NotSyncedUpErrorAllOfFromJSON,
  Ht as NotSyncedUpErrorAllOfFromJSONTyped,
  So as NotSyncedUpErrorAllOfToJSON,
  go as NotSyncedUpErrorFromJSON,
  me as NotSyncedUpErrorFromJSONTyped,
  mo as NotSyncedUpErrorToJSON,
  Jo as PublicKeyEcdsaSecp256k1AllOfFromJSON,
  Yt as PublicKeyEcdsaSecp256k1AllOfFromJSONTyped,
  Io as PublicKeyEcdsaSecp256k1AllOfToJSON,
  ho as PublicKeyEcdsaSecp256k1FromJSON,
  pe as PublicKeyEcdsaSecp256k1FromJSONTyped,
  bo as PublicKeyEcdsaSecp256k1ToJSON,
  Ao as PublicKeyEddsaEd25519AllOfFromJSON,
  Zt as PublicKeyEddsaEd25519AllOfFromJSONTyped,
  qo as PublicKeyEddsaEd25519AllOfToJSON,
  vo as PublicKeyEddsaEd25519FromJSON,
  ye as PublicKeyEddsaEd25519FromJSONTyped,
  Eo as PublicKeyEddsaEd25519ToJSON,
  v as PublicKeyFromJSON,
  $ as PublicKeyFromJSONTyped,
  O as PublicKeyToJSON,
  Oo as PublicKeyType,
  Qt as PublicKeyTypeFromJSON,
  Xt as PublicKeyTypeFromJSONTyped,
  No as PublicKeyTypeToJSON,
  d as RequiredError,
  Pi as ResourceAggregationLevel,
  p as ResourceAggregationLevelFromJSON,
  ct as ResourceAggregationLevelFromJSONTyped,
  xi as ResourceAggregationLevelToJSON,
  Be as ResponseError,
  xo as ResultSetCursorMixinFromJSON,
  jt as ResultSetCursorMixinFromJSONTyped,
  Co as ResultSetCursorMixinToJSON,
  en as ScryptoSborValueFromJSON,
  tn as ScryptoSborValueFromJSONTyped,
  nn as ScryptoSborValueToJSON,
  Ml as StateApi,
  Ko as StateEntityDetailsRequestAllOfFromJSON,
  on as StateEntityDetailsRequestAllOfFromJSONTyped,
  Go as StateEntityDetailsRequestAllOfToJSON,
  Mo as StateEntityDetailsRequestFromJSON,
  rn as StateEntityDetailsRequestFromJSONTyped,
  sn as StateEntityDetailsRequestToJSON,
  Yo as StateEntityDetailsResponseAllOfFromJSON,
  pn as StateEntityDetailsResponseAllOfFromJSONTyped,
  Zo as StateEntityDetailsResponseAllOfToJSON,
  ra as StateEntityDetailsResponseComponentDetailsAllOfFromJSON,
  yn as StateEntityDetailsResponseComponentDetailsAllOfFromJSONTyped,
  ia as StateEntityDetailsResponseComponentDetailsAllOfToJSON,
  ea as StateEntityDetailsResponseComponentDetailsFromJSON,
  Ne as StateEntityDetailsResponseComponentDetailsFromJSONTyped,
  ta as StateEntityDetailsResponseComponentDetailsToJSON,
  gn as StateEntityDetailsResponseFromJSON,
  mn as StateEntityDetailsResponseFromJSONTyped,
  la as StateEntityDetailsResponseFungibleResourceDetailsAllOfFromJSON,
  Sn as StateEntityDetailsResponseFungibleResourceDetailsAllOfFromJSONTyped,
  ca as StateEntityDetailsResponseFungibleResourceDetailsAllOfToJSON,
  oa as StateEntityDetailsResponseFungibleResourceDetailsFromJSON,
  Re as StateEntityDetailsResponseFungibleResourceDetailsFromJSONTyped,
  aa as StateEntityDetailsResponseFungibleResourceDetailsToJSON,
  _a as StateEntityDetailsResponseFungibleVaultDetailsFromJSON,
  Fe as StateEntityDetailsResponseFungibleVaultDetailsFromJSONTyped,
  fa as StateEntityDetailsResponseFungibleVaultDetailsToJSON,
  an as StateEntityDetailsResponseItemAncestorIdentitiesFromJSON,
  un as StateEntityDetailsResponseItemAncestorIdentitiesFromJSONTyped,
  ln as StateEntityDetailsResponseItemAncestorIdentitiesToJSON,
  _n as StateEntityDetailsResponseItemDetailsFromJSON,
  F as StateEntityDetailsResponseItemDetailsFromJSONTyped,
  h as StateEntityDetailsResponseItemDetailsToJSON,
  $o as StateEntityDetailsResponseItemDetailsType,
  cn as StateEntityDetailsResponseItemDetailsTypeFromJSON,
  dn as StateEntityDetailsResponseItemDetailsTypeFromJSONTyped,
  Bo as StateEntityDetailsResponseItemDetailsTypeToJSON,
  Se as StateEntityDetailsResponseItemFromJSON,
  fn as StateEntityDetailsResponseItemFromJSONTyped,
  Oe as StateEntityDetailsResponseItemToJSON,
  Sa as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfFromJSON,
  On as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfFromJSONTyped,
  Oa as StateEntityDetailsResponseNonFungibleResourceDetailsAllOfToJSON,
  ma as StateEntityDetailsResponseNonFungibleResourceDetailsFromJSON,
  he as StateEntityDetailsResponseNonFungibleResourceDetailsFromJSONTyped,
  pa as StateEntityDetailsResponseNonFungibleResourceDetailsToJSON,
  Ra as StateEntityDetailsResponseNonFungibleVaultDetailsFromJSON,
  be as StateEntityDetailsResponseNonFungibleVaultDetailsFromJSONTyped,
  Fa as StateEntityDetailsResponseNonFungibleVaultDetailsToJSON,
  Ia as StateEntityDetailsResponsePackageDetailsAllOfFromJSON,
  Nn as StateEntityDetailsResponsePackageDetailsAllOfFromJSONTyped,
  wa as StateEntityDetailsResponsePackageDetailsAllOfToJSON,
  ba as StateEntityDetailsResponsePackageDetailsFromJSON,
  Te as StateEntityDetailsResponsePackageDetailsFromJSONTyped,
  Ta as StateEntityDetailsResponsePackageDetailsToJSON,
  Qo as StateEntityDetailsResponseToJSON,
  Aa as StateEntityFungibleResourceVaultsPageRequestAllOfFromJSON,
  hn as StateEntityFungibleResourceVaultsPageRequestAllOfFromJSONTyped,
  qa as StateEntityFungibleResourceVaultsPageRequestAllOfToJSON,
  Ea as StateEntityFungibleResourceVaultsPageRequestFromJSON,
  Rn as StateEntityFungibleResourceVaultsPageRequestFromJSONTyped,
  Fn as StateEntityFungibleResourceVaultsPageRequestToJSON,
  bn as StateEntityFungibleResourceVaultsPageResponseFromJSON,
  Tn as StateEntityFungibleResourceVaultsPageResponseFromJSONTyped,
  xa as StateEntityFungibleResourceVaultsPageResponseToJSON,
  Ma as StateEntityFungiblesPageRequestAllOfFromJSON,
  wn as StateEntityFungiblesPageRequestAllOfFromJSONTyped,
  La as StateEntityFungiblesPageRequestAllOfToJSON,
  Va as StateEntityFungiblesPageRequestFromJSON,
  Jn as StateEntityFungiblesPageRequestFromJSONTyped,
  In as StateEntityFungiblesPageRequestToJSON,
  vn as StateEntityFungiblesPageResponseFromJSON,
  En as StateEntityFungiblesPageResponseFromJSONTyped,
  Ga as StateEntityFungiblesPageResponseToJSON,
  za as StateEntityMetadataPageRequestAllOfFromJSON,
  qn as StateEntityMetadataPageRequestAllOfFromJSONTyped,
  Wa as StateEntityMetadataPageRequestAllOfToJSON,
  $a as StateEntityMetadataPageRequestFromJSON,
  Dn as StateEntityMetadataPageRequestFromJSONTyped,
  An as StateEntityMetadataPageRequestToJSON,
  Pn as StateEntityMetadataPageResponseFromJSON,
  xn as StateEntityMetadataPageResponseFromJSONTyped,
  Qa as StateEntityMetadataPageResponseToJSON,
  ja as StateEntityNonFungibleIdsPageRequestAllOfFromJSON,
  kn as StateEntityNonFungibleIdsPageRequestAllOfFromJSONTyped,
  eu as StateEntityNonFungibleIdsPageRequestAllOfToJSON,
  Ya as StateEntityNonFungibleIdsPageRequestFromJSON,
  Cn as StateEntityNonFungibleIdsPageRequestFromJSONTyped,
  Vn as StateEntityNonFungibleIdsPageRequestToJSON,
  Mn as StateEntityNonFungibleIdsPageResponseFromJSON,
  Ln as StateEntityNonFungibleIdsPageResponseFromJSONTyped,
  nu as StateEntityNonFungibleIdsPageResponseToJSON,
  iu as StateEntityNonFungibleResourceVaultsPageRequestFromJSON,
  Kn as StateEntityNonFungibleResourceVaultsPageRequestFromJSONTyped,
  Gn as StateEntityNonFungibleResourceVaultsPageRequestToJSON,
  Un as StateEntityNonFungibleResourceVaultsPageResponseFromJSON,
  $n as StateEntityNonFungibleResourceVaultsPageResponseFromJSONTyped,
  ou as StateEntityNonFungibleResourceVaultsPageResponseToJSON,
  uu as StateEntityNonFungiblesPageRequestFromJSON,
  Bn as StateEntityNonFungiblesPageRequestFromJSONTyped,
  zn as StateEntityNonFungiblesPageRequestToJSON,
  Wn as StateEntityNonFungiblesPageResponseFromJSON,
  Hn as StateEntityNonFungiblesPageResponseFromJSONTyped,
  cu as StateEntityNonFungiblesPageResponseToJSON,
  gu as StateNonFungibleDataRequestAllOfFromJSON,
  Yn as StateNonFungibleDataRequestAllOfFromJSONTyped,
  mu as StateNonFungibleDataRequestAllOfToJSON,
  _u as StateNonFungibleDataRequestFromJSON,
  Qn as StateNonFungibleDataRequestFromJSONTyped,
  Xn as StateNonFungibleDataRequestToJSON,
  Nu as StateNonFungibleDataResponseAllOfFromJSON,
  tr as StateNonFungibleDataResponseAllOfFromJSONTyped,
  Ru as StateNonFungibleDataResponseAllOfToJSON,
  jn as StateNonFungibleDataResponseFromJSON,
  er as StateNonFungibleDataResponseFromJSONTyped,
  Su as StateNonFungibleDataResponseToJSON,
  Je as StateNonFungibleDetailsResponseItemFromJSON,
  Zn as StateNonFungibleDetailsResponseItemFromJSONTyped,
  Ie as StateNonFungibleDetailsResponseItemToJSON,
  Tu as StateNonFungibleIdsRequestAllOfFromJSON,
  ir as StateNonFungibleIdsRequestAllOfFromJSONTyped,
  Ju as StateNonFungibleIdsRequestAllOfToJSON,
  hu as StateNonFungibleIdsRequestFromJSON,
  nr as StateNonFungibleIdsRequestFromJSONTyped,
  rr as StateNonFungibleIdsRequestToJSON,
  Eu as StateNonFungibleIdsResponseAllOfFromJSON,
  ar as StateNonFungibleIdsResponseAllOfFromJSONTyped,
  Du as StateNonFungibleIdsResponseAllOfToJSON,
  sr as StateNonFungibleIdsResponseFromJSON,
  or as StateNonFungibleIdsResponseFromJSONTyped,
  wu as StateNonFungibleIdsResponseToJSON,
  xu as StateValidatorsListRequestAllOfFromJSON,
  cr as StateValidatorsListRequestAllOfFromJSONTyped,
  Cu as StateValidatorsListRequestAllOfToJSON,
  qu as StateValidatorsListRequestFromJSON,
  ur as StateValidatorsListRequestFromJSONTyped,
  lr as StateValidatorsListRequestToJSON,
  Uu as StateValidatorsListResponseAllOfFromJSON,
  Sr as StateValidatorsListResponseAllOfFromJSONTyped,
  $u as StateValidatorsListResponseAllOfToJSON,
  pr as StateValidatorsListResponseFromJSON,
  yr as StateValidatorsListResponseFromJSONTyped,
  Ku as StateValidatorsListResponseToJSON,
  Ll as StatusApi,
  Kl as StreamApi,
  Zu as StreamTransactionsRequestAllOfFromJSON,
  Rr as StreamTransactionsRequestAllOfFromJSONTyped,
  Qu as StreamTransactionsRequestAllOfKindFilterEnum,
  Xu as StreamTransactionsRequestAllOfOrderEnum,
  ju as StreamTransactionsRequestAllOfToJSON,
  Hu as StreamTransactionsRequestFromJSON,
  Or as StreamTransactionsRequestFromJSONTyped,
  Bu as StreamTransactionsRequestKindFilterEnum,
  zu as StreamTransactionsRequestOrderEnum,
  Nr as StreamTransactionsRequestToJSON,
  Fr as StreamTransactionsResponseFromJSON,
  hr as StreamTransactionsResponseFromJSONTyped,
  tl as StreamTransactionsResponseToJSON,
  oi as TextApiResponse,
  Qe as TokenAmountFromJSON,
  Xe as TokenAmountFromJSONTyped,
  Ye as TokenAmountToJSON,
  Gl as TransactionApi,
  rl as TransactionCommittedDetailsRequestFromJSON,
  br as TransactionCommittedDetailsRequestFromJSONTyped,
  Tr as TransactionCommittedDetailsRequestToJSON,
  ul as TransactionCommittedDetailsResponseAllOfFromJSON,
  vr as TransactionCommittedDetailsResponseAllOfFromJSONTyped,
  ll as TransactionCommittedDetailsResponseAllOfToJSON,
  Ae as TransactionCommittedDetailsResponseDetailsFromJSON,
  Jr as TransactionCommittedDetailsResponseDetailsFromJSONTyped,
  qe as TransactionCommittedDetailsResponseDetailsToJSON,
  Ir as TransactionCommittedDetailsResponseFromJSON,
  wr as TransactionCommittedDetailsResponseFromJSONTyped,
  ol as TransactionCommittedDetailsResponseToJSON,
  Er as TransactionConstructionResponseFromJSON,
  Dr as TransactionConstructionResponseFromJSONTyped,
  dl as TransactionConstructionResponseToJSON,
  fl as TransactionNotFoundErrorFromJSON,
  Pe as TransactionNotFoundErrorFromJSONTyped,
  gl as TransactionNotFoundErrorToJSON,
  Ar as TransactionPreviewRequestFlagsFromJSON,
  qr as TransactionPreviewRequestFlagsFromJSONTyped,
  Pr as TransactionPreviewRequestFlagsToJSON,
  yl as TransactionPreviewRequestFromJSON,
  xr as TransactionPreviewRequestFromJSONTyped,
  Cr as TransactionPreviewRequestToJSON,
  Lr as TransactionPreviewResponseFromJSON,
  Kr as TransactionPreviewResponseFromJSONTyped,
  Vr as TransactionPreviewResponseLogsInnerFromJSON,
  kr as TransactionPreviewResponseLogsInnerFromJSONTyped,
  Mr as TransactionPreviewResponseLogsInnerToJSON,
  Nl as TransactionPreviewResponseToJSON,
  _i as TransactionStatus,
  T as TransactionStatusFromJSON,
  Ze as TransactionStatusFromJSONTyped,
  bl as TransactionStatusRequestAllOfFromJSON,
  $r as TransactionStatusRequestAllOfFromJSONTyped,
  Tl as TransactionStatusRequestAllOfToJSON,
  Fl as TransactionStatusRequestFromJSON,
  Gr as TransactionStatusRequestFromJSONTyped,
  Ur as TransactionStatusRequestToJSON,
  El as TransactionStatusResponseAllOfFromJSON,
  Hr as TransactionStatusResponseAllOfFromJSONTyped,
  Dl as TransactionStatusResponseAllOfToJSON,
  zr as TransactionStatusResponseFromJSON,
  Wr as TransactionStatusResponseFromJSONTyped,
  xe as TransactionStatusResponseKnownPayloadItemFromJSON,
  Br as TransactionStatusResponseKnownPayloadItemFromJSONTyped,
  Ce as TransactionStatusResponseKnownPayloadItemToJSON,
  wl as TransactionStatusResponseToJSON,
  fi as TransactionStatusToJSON,
  ql as TransactionSubmitRequestFromJSON,
  Qr as TransactionSubmitRequestFromJSONTyped,
  Xr as TransactionSubmitRequestToJSON,
  Yr as TransactionSubmitResponseFromJSON,
  Zr as TransactionSubmitResponseFromJSONTyped,
  xl as TransactionSubmitResponseToJSON,
  se as ValidationErrorsAtPathFromJSON,
  It as ValidationErrorsAtPathFromJSONTyped,
  oe as ValidationErrorsAtPathToJSON,
  Vl as ValidatorCollectionAllOfFromJSON,
  jr as ValidatorCollectionAllOfFromJSONTyped,
  kl as ValidatorCollectionAllOfToJSON,
  Ee as ValidatorCollectionFromJSON,
  mr as ValidatorCollectionFromJSONTyped,
  dr as ValidatorCollectionItemActiveInEpochFromJSON,
  _r as ValidatorCollectionItemActiveInEpochFromJSONTyped,
  fr as ValidatorCollectionItemActiveInEpochToJSON,
  we as ValidatorCollectionItemFromJSON,
  gr as ValidatorCollectionItemFromJSONTyped,
  ve as ValidatorCollectionItemToJSON,
  De as ValidatorCollectionToJSON,
  ii as VoidApiResponse,
  ri as canConsumeForm,
  n as exists,
  ui as instanceOfAtLedgerStateMixin,
  gi as instanceOfCommittedTransactionInfo,
  mi as instanceOfCursorLimitMixin,
  Ni as instanceOfEntityMetadataCollection,
  Ri as instanceOfEntityMetadataCollectionAllOf,
  Oi as instanceOfEntityMetadataItem,
  Si as instanceOfEntityMetadataItemValue,
  bi as instanceOfEntityMetadataItemValueAllOf,
  wi as instanceOfEntityNotFoundError,
  Di as instanceOfErrorResponse,
  Vi as instanceOfFungibleResourcesCollection,
  ki as instanceOfFungibleResourcesCollectionAllOf,
  Ci as instanceOfFungibleResourcesCollectionItem,
  Ki as instanceOfFungibleResourcesCollectionItemGloballyAggregated,
  Gi as instanceOfFungibleResourcesCollectionItemGloballyAggregatedAllOf,
  Wi as instanceOfFungibleResourcesCollectionItemVaultAggregated,
  Xi as instanceOfFungibleResourcesCollectionItemVaultAggregatedAllOf,
  zi as instanceOfFungibleResourcesCollectionItemVaultAggregatedVault,
  ji as instanceOfFungibleResourcesCollectionItemVaultAggregatedVaultAllOf,
  Bi as instanceOfFungibleResourcesCollectionItemVaultAggregatedVaultItem,
  Ii as instanceOfGatewayError,
  ns as instanceOfGatewayInfoResponseKnownTarget,
  ss as instanceOfGatewayInfoResponseReleaseInfo,
  as as instanceOfGatewayStatusResponse,
  ls as instanceOfGatewayStatusResponseAllOf,
  _s as instanceOfInternalServerError,
  ms as instanceOfInternalServerErrorAllOf,
  Ss as instanceOfInvalidEntityError,
  Fs as instanceOfInvalidRequestError,
  Ts as instanceOfInvalidRequestErrorAllOf,
  ws as instanceOfInvalidTransactionError,
  os as instanceOfLedgerState,
  Ds as instanceOfLedgerStateMixin,
  ai as instanceOfLedgerStateSelector,
  xs as instanceOfNetworkConfigurationResponse,
  Ps as instanceOfNetworkConfigurationResponseWellKnownAddresses,
  Ls as instanceOfNonFungibleIdsCollection,
  Ks as instanceOfNonFungibleIdsCollectionAllOf,
  Ms as instanceOfNonFungibleIdsCollectionItem,
  Bs as instanceOfNonFungibleResourcesCollection,
  zs as instanceOfNonFungibleResourcesCollectionAllOf,
  $s as instanceOfNonFungibleResourcesCollectionItem,
  Qs as instanceOfNonFungibleResourcesCollectionItemGloballyAggregated,
  Zs as instanceOfNonFungibleResourcesCollectionItemGloballyAggregatedAllOf,
  ro as instanceOfNonFungibleResourcesCollectionItemVaultAggregated,
  oo as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedAllOf,
  no as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVault,
  lo as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVaultAllOf,
  to as instanceOfNonFungibleResourcesCollectionItemVaultAggregatedVaultItem,
  fo as instanceOfNotSyncedUpError,
  po as instanceOfNotSyncedUpErrorAllOf,
  Ro as instanceOfPublicKey,
  Fo as instanceOfPublicKeyEcdsaSecp256k1,
  To as instanceOfPublicKeyEcdsaSecp256k1AllOf,
  wo as instanceOfPublicKeyEddsaEd25519,
  Do as instanceOfPublicKeyEddsaEd25519AllOf,
  Po as instanceOfResultSetCursorMixin,
  Vo as instanceOfScryptoSborValue,
  ko as instanceOfStateEntityDetailsRequest,
  Lo as instanceOfStateEntityDetailsRequestAllOf,
  Ho as instanceOfStateEntityDetailsResponse,
  Xo as instanceOfStateEntityDetailsResponseAllOf,
  jo as instanceOfStateEntityDetailsResponseComponentDetails,
  na as instanceOfStateEntityDetailsResponseComponentDetailsAllOf,
  sa as instanceOfStateEntityDetailsResponseFungibleResourceDetails,
  ua as instanceOfStateEntityDetailsResponseFungibleResourceDetailsAllOf,
  da as instanceOfStateEntityDetailsResponseFungibleVaultDetails,
  Wo as instanceOfStateEntityDetailsResponseItem,
  Uo as instanceOfStateEntityDetailsResponseItemAncestorIdentities,
  zo as instanceOfStateEntityDetailsResponseItemDetails,
  ga as instanceOfStateEntityDetailsResponseNonFungibleResourceDetails,
  ya as instanceOfStateEntityDetailsResponseNonFungibleResourceDetailsAllOf,
  Na as instanceOfStateEntityDetailsResponseNonFungibleVaultDetails,
  ha as instanceOfStateEntityDetailsResponsePackageDetails,
  Ja as instanceOfStateEntityDetailsResponsePackageDetailsAllOf,
  va as instanceOfStateEntityFungibleResourceVaultsPageRequest,
  Da as instanceOfStateEntityFungibleResourceVaultsPageRequestAllOf,
  Pa as instanceOfStateEntityFungibleResourceVaultsPageResponse,
  Ca as instanceOfStateEntityFungiblesPageRequest,
  ka as instanceOfStateEntityFungiblesPageRequestAllOf,
  Ka as instanceOfStateEntityFungiblesPageResponse,
  Ua as instanceOfStateEntityMetadataPageRequest,
  Ba as instanceOfStateEntityMetadataPageRequestAllOf,
  Ha as instanceOfStateEntityMetadataPageResponse,
  Xa as instanceOfStateEntityNonFungibleIdsPageRequest,
  Za as instanceOfStateEntityNonFungibleIdsPageRequestAllOf,
  tu as instanceOfStateEntityNonFungibleIdsPageResponse,
  ru as instanceOfStateEntityNonFungibleResourceVaultsPageRequest,
  su as instanceOfStateEntityNonFungibleResourceVaultsPageResponse,
  au as instanceOfStateEntityNonFungiblesPageRequest,
  lu as instanceOfStateEntityNonFungiblesPageResponse,
  du as instanceOfStateNonFungibleDataRequest,
  fu as instanceOfStateNonFungibleDataRequestAllOf,
  yu as instanceOfStateNonFungibleDataResponse,
  Ou as instanceOfStateNonFungibleDataResponseAllOf,
  pu as instanceOfStateNonFungibleDetailsResponseItem,
  Fu as instanceOfStateNonFungibleIdsRequest,
  bu as instanceOfStateNonFungibleIdsRequestAllOf,
  Iu as instanceOfStateNonFungibleIdsResponse,
  vu as instanceOfStateNonFungibleIdsResponseAllOf,
  Au as instanceOfStateValidatorsListRequest,
  Pu as instanceOfStateValidatorsListRequestAllOf,
  Lu as instanceOfStateValidatorsListResponse,
  Gu as instanceOfStateValidatorsListResponseAllOf,
  Wu as instanceOfStreamTransactionsRequest,
  Yu as instanceOfStreamTransactionsRequestAllOf,
  el as instanceOfStreamTransactionsResponse,
  di as instanceOfTokenAmount,
  nl as instanceOfTransactionCommittedDetailsRequest,
  sl as instanceOfTransactionCommittedDetailsResponse,
  al as instanceOfTransactionCommittedDetailsResponseAllOf,
  il as instanceOfTransactionCommittedDetailsResponseDetails,
  cl as instanceOfTransactionConstructionResponse,
  _l as instanceOfTransactionNotFoundError,
  pl as instanceOfTransactionPreviewRequest,
  ml as instanceOfTransactionPreviewRequestFlags,
  Ol as instanceOfTransactionPreviewResponse,
  Sl as instanceOfTransactionPreviewResponseLogsInner,
  Rl as instanceOfTransactionStatusRequest,
  hl as instanceOfTransactionStatusRequestAllOf,
  Il as instanceOfTransactionStatusResponse,
  vl as instanceOfTransactionStatusResponseAllOf,
  Jl as instanceOfTransactionStatusResponseKnownPayloadItem,
  Al as instanceOfTransactionSubmitRequest,
  Pl as instanceOfTransactionSubmitResponse,
  Rs as instanceOfValidationErrorsAtPath,
  Mu as instanceOfValidatorCollection,
  Cl as instanceOfValidatorCollectionAllOf,
  ku as instanceOfValidatorCollectionItem,
  Vu as instanceOfValidatorCollectionItemActiveInEpoch,
  ni as mapValues,
  z as querystring
};
